### PR TITLE
added anti-aliasing support and some documentation

### LINKED ISSUE #186: Add toggle for Anti-Aliasing
Some games such as pixel or 8-bit games don't benefit from anti-aliasing as it makes the assets look blurry. We can add a flag in `Util::Image` to toggle between `GL_LINEAR` and `GL_NEAREST`

### PR DESCRIPTION
**Pull request by NOOBDY**
Resolves #186

### CODE REVIEW

**Review by Dragonfly911117 — APPROVED** [review]
WOMM

File: include/Util/Image.hpp [hunk 1]
```diff
@@ -29,8 +28,10 @@ class Image : public Core::Drawable {
      * @brief Constructor that takes a file path to the image.
      *
      * @param filepath The file path to the image.
+     * @param useAA Flag indicating whether anti-aliasing should be enabled
+     * (default is true).
      */
-    explicit Image(const std::string &filepath);
+    Image(const std::string &filepath, bool useAA = true);
```
Comment (ntut-rick) [inline include/Util/Image.hpp:34]: why do you remove the `explicit`

**Review by ntut-rick — APPROVED** [review]
LGTM

**Comment by Dragonfly911117** [comment]
Why am I asked to review this again? Thought I've already approved this.

File: include/Util/Image.hpp (hunk 1 above)
Comment (NOOBDY) [reply include/Util/Image.hpp:34]: because there are 2 parameters and C++ can't implicit convert in this case

**Pull request merged**